### PR TITLE
Remove unused functions

### PR DESCRIPTION
**Pull request by qarkai**
Found by [xunused](https://github.com/mgehre/xunused).
Skipped const counterparts and some other overloads.

### CODE REVIEW

File: src/fheroes2/game/game_newgame.cpp [hunk 1]
```diff
@@ -269,50 +269,6 @@ fheroes2::GameMode Game::NewPriceOfLoyaltyCampaign()
     return gameChoice;
 }
 
-fheroes2::GameMode Game::NewNetwork()
```
Comment (ihhub) [inline src/fheroes2/game/game_newgame.cpp:272]: Please revert this change. We'll need it for the future.

File: src/fheroes2/monster/monster_info.cpp [hunk 2]
```diff
@@ -605,54 +605,6 @@ namespace fheroes2
         return "";
     }
 
-    std::string getMonsterDescription( const int monsterId )
```
Comment (ihhub) [inline src/fheroes2/monster/monster_info.cpp:608]: Please rever this change. It's useful for tools.

File: src/fheroes2/ai/ai_base.cpp [hunk 3]
```diff
@@ -37,16 +37,6 @@ namespace AI
         return "base";
     }
 
-    const char * Base::License() const
-    {
-        return "GPL-2.0";
-    }
-
-    int Base::GetPersonality() const
```
Comment (ihhub) [inline src/fheroes2/ai/ai_base.cpp:45]: Please revert this method. We are going to add personality very soon.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @qarkai , I left several remarks here. Could you please take a look?

File: src/fheroes2/monster/monster_info.cpp (hunk 2 above)
Comment (qarkai) [reply src/fheroes2/monster/monster_info.cpp]: But there is no tools which use this functions :)

**Comment by qarkai** [comment]
Those functions could be restored on demand but whatever.

File: src/fheroes2/monster/monster_info.cpp (hunk 2 above)
Comment (ihhub) [reply src/fheroes2/monster/monster_info.cpp]: For my usage and tools :)

**Comment by ihhub** [comment]
@qarkai , huge thanks for code cleanup!

**Pull request merged**